spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/JacksonModelAttributeSnippet.java

```diff
@@ -59,17 +60,15 @@ public JacksonModelAttributeSnippet(Collection<HandlerMethodArgumentResolver> ha
     }
 
     @Override
-    protected Type getType(HandlerMethod method) {
-        for (MethodParameter param : method.getMethodParameters()) {
-            if (isModelAttribute(param) || isProcessedAsModelAttribute(param)) {
-                return getType(param);
-            }
-        }
-        return null;
+    protected Type[] getType(HandlerMethod method) {
+        return Arrays.stream(method.getMethodParameters())
+                .filter(param -> isModelAttribute(param) || isProcessedAsModelAttribute(param))
+                .map(this::getType)
+                .toArray(Type[]::new);
     }
 
     private boolean isModelAttribute(MethodParameter param) {
-        return param.getParameterAnnotation(ModelAttribute.class) != null;
+        return !BeanUtils.isSimpleProperty(param.getParameterType()) || param.getParameterAnnotation(ModelAttribute.class) != null;
```

Comment:
Are you sure this won't catch also everything else like `@RequestBody` DTOs? We should test it on a real example having different annotations in a single controller. We might end up duplicating info.

Reply:
You're right. We have to check that the parameter is not annotated at all and that it's not a simple type

Reply:
fixed.